### DISCUSSION TITLE
Update code headers

### DISCUSSION THREAD

**Discussion by jonasraoni** [pkp/pkp-lib#7986] [Proposals]
**Describe the problem you would like to solve**
Our current code header requires some maintenance. Perhaps we can adopt another style to make it simpler to maintain.
There are many variants out there, even big companies have mixed styles across their codebase. Follow some examples:

- Only the starting year (Adobe, Google, Microsoft, Salesforce, IBM): https://github.com/microsoft/vscode/blob/main/LICENSE.txt
- Starting + last updated year (Apple, Oracle): https://github.com/mysql/mysql-server/blob/8.0/README
- No dates, only the authors (Microsoft): https://github.com/dotnet/aspnetcore/blob/main/LICENSE.txt
- No header at all, just a `LICENSE` file (GitHub)

**Describe the solution you'd like**
In case we have lawyers available, it might be worth to check with them if adding a "last updated date" is enough to extend the copyright (the repositories at https://github.com/sfu didn't give me any clues). If not, then I'd opt-in for the approach that includes only the starting year.

**Who is asking for this feature?**
Developer

**Comment by asmecher** [comment]
@jonasraoni, this is more about the code header than the license, right? The code headers don't contain the license (that's typically in `docs/COPYING` or `LICENSE`), but they do make reference to it, and also state copyright and some self-documentation (`@class`, `@ingroup`, `@brief`, etc).

**Reply by jonasraoni** [reply]
As we're following the Google/Microsoft style (only the starting year), I think we can write a script to capture when the file was created in Git and use it as the initial year, or something safer like `min(starting year, created year) - present` (to cover the files which were created before moving to GitHub), if agreed, I think this can be turned into an issue.

**Comment by jonasraoni** [comment]
Yes, specifically about the copyright part of the code header :)
